core/bones/password.py

```diff
@@ -69,21 +63,58 @@ class PasswordBone(StringBone):
         "core.bones.password.tooWeakMessage",
         defaultText="The entered password is too weak."
     )
+    _password_tests: List[Tuple] = [
+        (r"(?=.*[A-Z])", translate("core.bones.password.no_capital_letters",
+                                   defaultText="The password entered has no capital letters.")),
+        (r"(?=.*[a-z])", translate("core.bones.password.no_lowercase_letters",
+                                   defaultText="The password entered has no lowercase letters.")),
+        (r"(?=.*\d)", translate("core.bones.password.no_digits",
+                                defaultText="The password entered has no digits.")),
+        (r"(?=.*\W)", translate("core.bones.password.no_special_characters",
```

Comment:
Related to the `re.search` --> `re.match` change, the REs should be:

```suggestion
        (r"^(?=.*[A-Z].*)$", translate("core.bones.password.no_capital_letters",
                                   defaultText="The password entered has no capital letters.")),
        (r"^(?=.*[a-z].*)$", translate("core.bones.password.no_lowercase_letters",
                                   defaultText="The password entered has no lowercase letters.")),
        (r"^(?=.*\d.*)$", translate("core.bones.password.no_digits",
                                defaultText="The password entered has no digits.")),
        (r"^(?=.*\W.*)$", translate("core.bones.password.no_special_characters",
```

Do we need the non-capturing-groups?

---

Furthermore I think it's more efficient to use a compiled version of the REs (`re.compile()`). We can use them several times, but compile them only once.
But I'm unsure how to store these.
If we store them directly in `self.password_tests`, the renderer must create a new list (`[test.pattern, hint for test, hint in self.passsword_tests`), which isn't efficient too.
Maybe we could internal a second list with the compiled versions.